code/modules/projectiles/projectile/experimental.dm

```diff
@@ -0,0 +1,673 @@
+	if((x == 1 || x == world.maxx || y == 1 || y == world.maxy))
+		//del(src)
+		OnDeath()
+		loc = null
+		returnToPool(src)
+		return
+	if(loc != starting)
+		var/obj/structure/ricochet_trail/trail = new(loc)
+		switch(pos_to)
+			if("N")
+				if(pos_from == "W")
+					trail.dir = NORTH
+				else
+					trail.dir = EAST
+			if("S")
+				if(pos_from == "W")
+					trail.dir = WEST
+				else
+					trail.dir = SOUTH
+			if("E")
+				if(pos_from == "N")
+					trail.dir = EAST
+				else
+					trail.dir = SOUTH
+			if("W")
+				if(pos_from == "N")
+					trail.dir = NORTH
+				else
+					trail.dir = WEST
+	step_towards(src, current)
+	if(!bumped)
+		switch(pos_to)
+			if("N")
+				if(pos_from == "W")
+					pos_to = "E"
+				else
+					pos_to = "W"
+				pos_from = "S"
+			if("S")
+				if(pos_from == "W")
+					pos_to = "E"
+				else
+					pos_to = "W"
+				pos_from = "N"
+			if("E")
+				if(pos_from == "N")
+					pos_to = "S"
+				else
+					pos_to = "N"
+				pos_from = "W"
+			if("W")
+				if(pos_from == "N")
+					pos_to = "S"
+				else
+					pos_to = "N"
+				pos_from = "E"
+	update_icon()
+	bumped = 0
+
+/obj/item/projectile/ricochet/proc/ricochet_jump()//movement through dense objects
+	var/to_x = x
+	var/to_y = y
+	switch(pos_to)
+		if("N")
+			to_y = y + 1
+		if("S")
+			to_y = y - 1
+		if("E")
+			to_x = x + 1
+		if("W")
+			to_x = x - 1
+	var/turf/T = locate(to_x, to_y, z)
+	if((x == 1 || x == world.maxx || y == 1 || y == world.maxy))
+		//del(src)
+		OnDeath()
+		loc = null
+		returnToPool(src)
+		return
+	var/obj/structure/ricochet_trail/trail = new(loc)
+	switch(pos_to)
+		if("N")
+			if(pos_from == "W")
+				trail.dir = NORTH
+			else
+				trail.dir = EAST
+		if("S")
+			if(pos_from == "W")
+				trail.dir = WEST
+			else
+				trail.dir = SOUTH
+		if("E")
+			if(pos_from == "N")
+				trail.dir = EAST
+			else
+				trail.dir = SOUTH
+		if("W")
+			if(pos_from == "N")
+				trail.dir = NORTH
+			else
+				trail.dir = WEST
+	loc = T
+	switch(pos_to)
+		if("N")
+			if(pos_from == "W")
+				pos_to = "E"
+			else
+				pos_to = "W"
+			pos_from = "S"
+		if("S")
+			if(pos_from == "W")
+				pos_to = "E"
+			else
+				pos_to = "W"
+			pos_from = "N"
+		if("E")
+			if(pos_from == "N")
+				pos_to = "S"
+			else
+				pos_to = "N"
+			pos_from = "W"
+		if("W")
+			if(pos_from == "N")
+				pos_to = "S"
+			else
+				pos_to = "N"
+			pos_from = "E"
+	update_icon()
+
+/obj/structure/ricochet_trail	//so pretty
+	name = "ricochet shot"
+	icon = 'icons/obj/projectiles_experimental.dmi'
+	icon_state = "ricochet"
+	opacity = 0
+	density = 0
+	unacidable = 1
+	anchored = 1
+	layer = 12
+
+/obj/structure/ricochet_trail/New()
+	. = ..()
+	spawn(30)
+		qdel(src)
+
+/obj/structure/ricochet_bump	//oh so pretty
+	name = "ricochet shot"
+	icon = 'icons/obj/projectiles_experimental.dmi'
+	icon_state = "ricochet_bounce"
+	opacity = 0
+	density = 0
+	unacidable = 1
+	anchored = 1
+	layer = 14
+
+/obj/structure/ricochet_bump/New()
+	. = ..()
+	spawn(30)
+		qdel(src)
```

Comment:
I tried replacing it entirely with proc/anim, but it had the same issue, I think it might be a Byond related issue, it only works properly when using structures.